src/integrationTest/java/com/mongodb/hibernate/query/mutate/AbstractMutateQueryIntegrationTests.java

```diff
@@ -0,0 +1,112 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.query.mutate;
+
+import static java.util.Collections.singleton;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.cfg.JdbcSettings.DIALECT;
+
+import com.mongodb.hibernate.dialect.MongoDialect;
+import com.mongodb.hibernate.query.AbstractQueryIntegrationTests;
+import java.util.Set;
+import java.util.function.Consumer;
+import org.hibernate.dialect.Dialect;
+import org.hibernate.engine.jdbc.dialect.spi.DialectResolutionInfo;
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.query.MutationQuery;
+import org.hibernate.sql.ast.SqlAstTranslator;
+import org.hibernate.sql.ast.SqlAstTranslatorFactory;
+import org.hibernate.sql.ast.tree.MutationStatement;
+import org.hibernate.sql.ast.tree.select.SelectStatement;
+import org.hibernate.sql.exec.spi.JdbcOperationQueryMutation;
+import org.hibernate.sql.exec.spi.JdbcOperationQuerySelect;
+import org.hibernate.sql.model.ast.TableMutation;
+import org.hibernate.sql.model.jdbc.JdbcMutationOperation;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.Setting;
+
+@ServiceRegistry(
+        settings = {
+            @Setting(
+                    name = DIALECT,
+                    value =
+                            "com.mongodb.hibernate.query.mutate.AbstractMutateQueryIntegrationTests$MutateTranslatorAwareDialect"),
+        })
+class AbstractMutateQueryIntegrationTests extends AbstractQueryIntegrationTests {
+
+    void assertAffectedTableNames(String mutateHql, String expectedAffectedTableName) {
+        assertAffectedTableNames(mutateHql, null, singleton(expectedAffectedTableName));
+    }
+
+    void assertAffectedTableNames(
+            String mutateHql, Consumer<MutationQuery> queryPostProcessor, String expectedAffectedTableName) {
+        assertAffectedTableNames(mutateHql, queryPostProcessor, singleton(expectedAffectedTableName));
+    }
+
+    void assertAffectedTableNames(
+            String mutateHql, Consumer<MutationQuery> queryPostProcessor, Set<String> expectedAffectedTableNames) {
+        getSessionFactoryScope().inTransaction(session -> {
+            var query = session.createMutationQuery(mutateHql);
+            if (queryPostProcessor != null) {
+                queryPostProcessor.accept(query);
+            }
+            query.executeUpdate();
+            var mutateTranslator =
+                    ((MutateTranslatorAwareDialect) session.getJdbcServices().getDialect()).getMutateSqlAstTranslator();
+            assertThat(mutateTranslator.getAffectedTableNames()).isEqualTo(expectedAffectedTableNames);
+        });
+    }
+
+    public static final class MutateTranslatorAwareDialect extends Dialect {
```

Comment:
An important feature to test is we need to include the table names involved in DML queries into the result of `SqlAstTranslator#getAffectedTableNames()` which acts as auto-flush purpose in the upper level caller of the translator.

We might need to finish other features (like table joining) to test the auto-flush features in integration test; currently we could only test this important feature by some expedient.